### PR TITLE
Improve performance of broadcasting in elementwise kernels

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

**Comment by okuta** [comment]
jenkins, test this please.

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -1482,6 +1482,19 @@ cdef class ndarray:
...
+                       const vector.vector[Py_ssize_t]& strides,
+                       bint update_c_contiguity,
+                       bint update_f_contiguity):
+        cdef ndarray v
+        v = ndarray.__new__(ndarray)
+        v.data = self.data
+        v.base = self.base if self.base is not None else self
+        v.dtype = self.dtype
+        v._set_shape_and_strides(
+            shape, strides, update_c_contiguity, update_f_contiguity)
```
Comment (niboshi) [inline cupy/core/core.pyx]: As neither the contiguity flags are explicitly set, I think `update_c_contiguity` and `update_f_contiguity` should be both `True`.

File: cupy/core/_kernel.pyx [hunk 2]
```diff
@@ -507,6 +500,9 @@ cdef class ElementwiseKernel:
         """
         cdef function.Function kern
         cdef Py_ssize_t size
+        cdef int i
```
Comment (niboshi) [inline cupy/core/_kernel.pyx]: `i` should be `Py_ssize_t` becaues it comes from `self.nin`.
(Btw I don't think the explicit declaration is needed.)

File: cupy/core/core.pyx (hunk 1 above)
Comment (okuta) [reply cupy/core/core.pyx]: Thanks. I added setting `self`'s flags.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by niboshi** [comment]
How about using `_view()` in `view()`?

**Comment by okuta** [comment]
Thanks. I fixed.
jenkins, test this please.

**Comment by niboshi** [comment]
LGTM

**Comment by niboshi** [comment]
Hmm, it seems this PR includes two improvements.
What's in common? How can I change the title?

**Comment by okuta** [comment]
OK, I will split.

**Comment by okuta** [comment]
I changed the title.

**Comment by niboshi** [comment]
I don't think this PR improves the performance of `broadcast`.
How about something like this? :

"Improve performance of broadcasting in elementwise kernels"

**Comment by okuta** [comment]
Good!

**Comment by niboshi** [comment]
Thanks, LGTM

**Pull request merged**